CalendarAppTests/CalendarEventHandlerTests.swift

```diff
@@ -0,0 +1,329 @@
+        handler.addEventsFromArray([], tomorrow)
+        handler.addNewEvent(longEvent)
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [])
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [longEvent])
+        XCTAssertEqual(handler.getEventsForDate(Date()), [longEvent])
+        
+        handler.deleteEvent(longEvent)
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [])
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [])
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+    }
+    
+    func testDeleteMultipleEvents() {
+        let longEvent = SampleEvents.makeEvent(on: yesterday, withDays: 2, withHours: 3)
+        handler.addEventsFromArray([currentEvent, longEvent], Date())
+        handler.addEventsFromArray([earlyEvent, longEvent], yesterday)
+        
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent, longEvent])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent, longEvent])
+        
+        handler.deleteEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [longEvent])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent, longEvent])
+        
+        handler.deleteEvent(longEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent])
+    }
+    
+    func testAddEventAfterDelete() {
+        let event = Event()
+        handler.addEventsFromArray([currentEvent, event], Date())
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent, event])
+        handler.deleteEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event])
+
+        handler.addNewEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event, currentEvent])
+    }
+    
+    func testUpdateEvent() {
+        handler.addEventsFromArray([currentEvent], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.eventTitle = "hi"
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateNonexistentEvent() {
+        handler.addEventsFromArray([], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateEventToNewDate() {
+        handler.addEventsFromArray([currentEvent], Date())
+        handler.addEventsFromArray([], tomorrow)
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.startDate = currentEvent.startDate.advanced(by: TimeInterval(86400))
```

Comment:
For magic numbers like this I would have a `static let oneDayInSeconds = TimeInterval(86400)`